robotidy/transformers/SplitTooLongLine.py

```diff
@@ -0,0 +1,112 @@
+from itertools import tee
+
+from robot.api.parsing import (
+    ModelTransformer,
+    Token
+)
+
+
+def eol():
+    return Token(Token.EOL)
+
+
+class SplitTooLongLine(ModelTransformer):
+    def __init__(self, line_length: int = 120):
+        self.line_length = line_length
+
+    def visit_KeywordCall(self, node):  # noqa
+        if all(token.end_col_offset <= self.line_length for token in node.tokens[::-1]):
+            return node
+        if not node.get_tokens(Token.ARGUMENT):  # return if there are no arguments - nothing to split
+            return node
+        return self.split_keyword_call(node)
+
+    def split_keyword_call(self, node):
+        is_single_line = node.lineno == node.end_lineno
+
+        indent = node.tokens[0]
+        assignment = list(self.insert_seperator(node.get_tokens(Token.ASSIGN)))
+        keyword = node.get_token(Token.KEYWORD)
+
+        if is_single_line:
+            comment = node.get_tokens(Token.COMMENT)
+            comment_line = [indent] + comment + [eol()] if comment else []
+
+            head = comment_line + [indent] + assignment + [keyword]
+
+            tail = node.tokens[node.tokens.index(keyword) + 1:]
+            tail = list(filter(lambda t: t.type != Token.COMMENT, tail))
+
+            lines = []
+            for tokenized_line in self.pack_arguments(head, tail, indent):
+                lines.extend(tokenized_line)
+
+        else:
+            return node  # Not implemented yet
+
+        node.tokens = lines
+        return node
+
+    def pack_arguments(self, tokens_so_far, remaining_tokens, indent):
+        if not remaining_tokens:
+            return []
+
+        # Ensure we are in the correct place to pack
+        assert (not remaining_tokens or (
+            remaining_tokens[0].type == Token.SEPARATOR
+            and remaining_tokens[1].type == Token.ARGUMENT
+        ))
+        assert tokens_so_far[-1].type != Token.EOL
+
+        line = tokens_so_far[:]
+
+        for token, next_token in self.lookahead(remaining_tokens):
+            if token.type in (Token.SEPARATOR, Token.CONTINUATION, Token.EOL):
+                continue
+
+            next_tokens = (
+                [self.separator(), token, self.separator(), next_token]
```

Comment:
rather than function that create new instance of separator every time, you can create variable like so:
```
separator = Token(Token.SEPARATOR, self.formatting_config.space_count * ' ')
```
And use it in code. It doesn't matter if you use the same instance - the initial line numbers are -1 and robot api will recalculate positions etc anyway so it's safe to use the same instance.

Reply:
Good to know. I will optimize that later, when multiline kw splits are implemented